pkg/test/framework/components/csrctrl/kube.go

```diff
@@ -0,0 +1,111 @@
+// Copyright Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package csrctrl
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"io/ioutil"
+
+	environ "istio.io/istio/pkg/test/env"
+	"istio.io/istio/pkg/test/framework/components/cluster"
+	"istio.io/istio/pkg/test/framework/components/namespace"
+	"istio.io/istio/pkg/test/framework/resource"
+	kube2 "istio.io/istio/pkg/test/kube"
+	"istio.io/istio/pkg/test/scopes"
+	"istio.io/istio/pkg/test/util/tmpl"
+)
+
+const (
+	sigcaNamespace = "sig-ca-system"
+)
+
+var (
+	_ Instance  = &kubeComponent{}
+	_ io.Closer = &kubeComponent{}
+)
+
+type kubeComponent struct {
+	id      resource.ID
+	ns      namespace.Instance
+	cluster cluster.Cluster
+}
+
+func newKube(ctx resource.Context, cfg Config) (Instance, error) {
```

Comment:
I think we do not need a new component. This will make the test slower, and its more complex to maintain.

Instead, the test can just run the CSR controller directly from the go test code. The go test already has access to the cluster of course, and its just acting on k8s resources, so there is no need to run in cluster.

Reply:
Hi John,  suppose we should have multiple test that will use this CSR controller to sign cert back. If we put this into test code, it means every test needs to write this logic, right? using current way, it will make it a pluggable mode that each test can use it easily.    This will deployed as a seperate pod and will sign cert for the signer you defined.  Can you please explain why this will make test slower? Thanks.